applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -34,101 +35,170 @@ namespace Kratos
+        } (rParameters)
+    )
+    , mReformModelPartAtEachTimeStep(
+        [&] (Parameters& x) {
+            x.ValidateAndAssignDefaults(GetDefaultSettings());
+            return x["reform_model_part_at_each_time_step"].GetBool();
+        } (rParameters)
+    )
+    , mVisualizationScalarVariables(
+        [&] (Parameters& x) -> std::vector<const Variable<double>*> {
+            x.ValidateAndAssignDefaults(GetDefaultSettings());
+            std::vector<const Variable<double>*> aux_list;
+            FillVariablesList<double>(x, aux_list);
+            return aux_list;
+        } (rParameters)
+    )
+    , mVisualizationVectorVariables(
+        [&] (Parameters& x) -> std::vector<const Variable<array_1d<double,3>>*> {
+            x.ValidateAndAssignDefaults(GetDefaultSettings());
+            std::vector<const Variable<array_1d<double,3>>*> aux_list;
+            FillVariablesList<array_1d<double,3>>(x, aux_list);
+            return aux_list;
+        } (rParameters)
+    )
 
-    // Check visualization scalar variables
-    for (unsigned int i_var = 0; i_var < mVisualizationScalarVariables.size(); ++i_var){
-        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(mVisualizationScalarVariables[i_var], r_orig_node);
-    }
+{
+}
 
-    // Check visualization vector variables
-    for (unsigned int i_var = 0; i_var < mVisualizationVectorVariables.size(); ++i_var){
-        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(mVisualizationVectorVariables[i_var], r_orig_node);
-    }
+EmbeddedSkinVisualizationProcess::EmbeddedSkinVisualizationProcess(
+    Model& rModel,
+    Parameters rParameters)
+    : EmbeddedSkinVisualizationProcess(
+        [&] (Model& x, Parameters& y) -> ModelPart& {
+            y.ValidateAndAssignDefaults(GetDefaultSettings());
+            KRATOS_ERROR_IF(y["model_part_name"].GetString() == "") << "\'model_part_name\' is empty. Please provide the origin model part name." << std::endl;
+            return x.GetModelPart(y["model_part_name"].GetString());
+        } (rModel, rParameters),
+        [&] (Model& x, Parameters& y) -> ModelPart& {
+            y.ValidateAndAssignDefaults(GetDefaultSettings());
+            return CreateAndPrepareVisualizationModelPart(x, y);
+        } (rModel, rParameters),
+        rParameters)
+{
+}
 
-    // Check visualization component variables
-    for (unsigned int i_var = 0; i_var < mVisualizationComponentVariables.size(); ++i_var){
-        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(mVisualizationComponentVariables[i_var], r_orig_node);
+void EmbeddedSkinVisualizationProcess::ExecuteBeforeSolutionLoop()
+{
+    // If mesh update is not required (constant level set function) create the visualization mesh once
+    if (!mReformModelPartAtEachTimeStep) {
+        this->CreateVisualizationMesh();
     }
-
-    // Initialize the visualization mesh creation flag
-    mSetVisualizationMesh = true;
-
-    KRATOS_CATCH("");
 }
 
 void EmbeddedSkinVisualizationProcess::ExecuteInitializeSolutionStep()
```

Comment:
Do you think we can do a full copy of the process info here?

` mrVisualizationModelPart.GetProcessInfo() = mrModelPart.GetProcessInfo();`

In the potential flow cases we write stuff in the ProcessInfo like `MACH_INFINITY`, 'FREE_STREAM_VELOCITY', that is read when writing the values on the integration points (for instance, to compute the `PRESSURE_COEFFICIENT`, we use the `FREE_STREAM_VELOCITY`). With the full copy of the ProcessInfo(), the visualization model part contains this ifnromation to output these elemental variables. 

```
-skin_visualizaiton- model part
    Buffer Size : 1
    Number of tables : 0
    Number of sub model parts : 0
    Current solution step index : 0
    DOMAIN_SIZE : 2
    STABILIZATION_FACTOR : 1
    REFERENCE_CHORD : 1
    IS_RESTARTED : 0
    FLAG_VARIABLE : 1
    WAKE_ORIGIN : [3](-0.25,0,0)
    ROTATION_ANGLE : 5
    FREE_STREAM_MACH : 0.0294118
    FREE_STREAM_VELOCITY : [3](10,0,0)
    FREE_STREAM_DENSITY : 1
    SOUND_VELOCITY : 340
    HEAT_CAPACITY_RATIO : 1.4
    MACH_LIMIT : 0.94
    MACH_SQUARED_LIMIT : 3
    TIME : 1
    DELTA_TIME : 1
    STEP : 1
    PRINTED_STEP : 1

```

Reply:
> Do you think we can do a full copy of the process info here?
> 
More than a copy I'd make them to use the same. I remember that at some point we discussed about this but we finally decided to do it in this way. What I like of the current design is that the model parts are completely independent. Nonetheless, I see the point on having the same process info container.